win32ss/user/ntuser/winpos.c

```diff
@@ -1904,18 +1904,13 @@ co_WinPosSetWindowPos(
    }
    else if (WinPos.flags & SWP_SHOWWINDOW)
    {
-      if (Window->spwndOwner == NULL ||
-          !(Window->spwndOwner->style & WS_VISIBLE) ||
-          (Window->spwndOwner->ExStyle & WS_EX_TOOLWINDOW))
+      if ((Window->ExStyle & (WS_EX_APPWINDOW | WS_EX_TOOLWINDOW)) >= WS_EX_APPWINDOW ||
```

Comment:
For this first test, why not just looking at `(Window->ExStyle & WS_EX_APPWINDOW) == WS_EX_APPWINDOW`, since:
WS_EX_APPWINDOW is == 0x40000 ; WS_EX_TOOLWINDOW == 128, so your original test would be true only when at least the WS_EX_APPWINDOW bit is set? (so basically at this stage the test result doesn't really change even if you didn't look at the WS_EX_TOOLWINDOW part).

Reply:
Be careful. It is not `==` but `>=`.

Reply:
Yes I know, when you do `ExStyle & (WS_EX_APPWINDOW | WS_EX_TOOLWINDOW)` the obtained result will be either 0x40000 (just the appwindow bit set) or 0x80 (== 128 , just the toolwindow bit set), or the "sum" 0x40080 (both bits set), and then, you do your greater-or-equal comparison with WS_EX_APPWINDOW (which is 0x40000).
So the only way this test can be TRUE, is when the `ExStyle & (WS_EX_APPWINDOW | WS_EX_TOOLWINDOW)` is equal to 0x40000 or 0x40080, i.e. in both cases the appwindow bit is set, and basically you don't care about the toolwindow.
Hence, in my opinion, this test can just be simplified to checking whether the appwindow bit is set, as I proposed above.

Reply:
Simplified about `WS_EX_APPWINDOW`.

Reply:
@HBelusca, Nice Simplification. Thanks.